### PR TITLE
Add Newcline components to master

### PR DESCRIPTION
**Pull request by martin-neugebauer**

### CODE REVIEW

**Comment by martin-neugebauer** [comment]
@zuckerruebe Could you have a quick look? Thanks!

File: trnsysGUI/SaltTank.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import trnsysGUI.createSinglePipePortItem as _cspi
+import trnsysGUI.images as _img
+import trnsysGUI.massFlowSolver.networkModel as _mfn
+from trnsysGUI.BlockItem import BlockItem
+from trnsysGUI.MyQFileSystemModel import MyQFileSystemModel
+from trnsysGUI.MyQTreeView import MyQTreeView
+from trnsysGUI.massFlowSolver import InternalPiping, MassFlowNetworkContributorMixin
+
+
+class SaltTank(BlockItem, MassFlowNetworkContributorMixin):
```
Comment (zuckerruebe) [inline trnsysGUI/SaltTank.py:19]: Good idea with the base class. Let's also make it abstract, i.e. inherit from `_abc.ABC` (`import abc as _abc`) ["abc" = "Abstract base class"].

File: trnsysGUI/diagram/Decoder.py [hunk 2]
```diff
@@ -110,6 +115,10 @@ def object_hook(self, arr: tp.Mapping[str, tp.Mapping[str, tp.Any]]) -> tp.Any:
                         bl = Collector(
                             "Collector", self.editor.diagramView, displayName=i["BlockDisplayName"], loadedBlock=True
                         )
+                    elif i["BlockName"] in ("CentralReceiver", "CR"):
```
[the remark this answers is not in the document]
Comment (martin-neugebauer) [reply trnsysGUI/diagram/Decoder.py:118]: I had some names in my old branch, that I'm not happy with now. So I thought I'd include those to be able to open the old json files, but also give them better names now. I am, however, not really sure where exactly the block name is defined to begin with. I think it's from the default component name, right?

File: trnsysGUI/SaltTank.py (hunk 1 above)
Comment (martin-neugebauer) [reply trnsysGUI/SaltTank.py:19]: What changes would this imply for this specific case?

File: trnsysGUI/diagram/Decoder.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (zuckerruebe) [reply trnsysGUI/diagram/Decoder.py:118]: Not sure what you mean by default component name. If it's not a big problem, I'd rather only have one `BlockName` here: the new one...

[the remark this answers is not in the document]
Comment (martin-neugebauer) [reply trnsysGUI/diagram/Decoder.py:118]: OK, let me rephrase: Where is the block name of a component defined?

[the remark this answers is not in the document]
Comment (zuckerruebe) [reply trnsysGUI/diagram/Decoder.py:118]: There is no one place. It's defined/referred to in multiple places (which is bad and should be improved). That's why it's a lot nicer to just have one block name.

[the remark this answers is not in the document]
Comment (martin-neugebauer) [reply trnsysGUI/diagram/Decoder.py:118]: That's what I meant and suspected.

**Comment by zuckerruebe** [comment]
Very much sharpened the "base" character there! If you're happy with it, feel free to hit that merge button, baby!
